### ISSUE TITLE
Table does not escape column line breaks

### ISSUE THREAD

**Issue by Normvariante** [BernieWhite/PSDocs#11]
Hi,

I noticed that if I convert an object that has a string with a line break in one of its properties, the line break is not escaped when converting it to a md table.

    $HasNewline = @"
    One, Two,
    Three
    "@

    $Object = [PsCustomObject] @{

        String1 = "1 2 3 4 5"
        String2 = $HasNewline
        String3 = "6 7 8 9 0"

    }

    Document Sample {

        $Object | Table
   
    }

    Invoke-PSDocument -Name 'Sample' -PassThru

The result is

    |String1|String2|String3|
    | --- | --- | --- |
    |1 2 3 4 5|One, Two,
    Three|6 7 8 9 0|

Which breaks the table if I render the markdown as HTML. The new lines should be escaped with `<br>` or `\`

**Comment by BernieWhite** [comment]
Thanks for reporting the issue. I assume this is using v0.3.0 of PSDocs?

**Comment by Normvariante** [comment]
Yes, I tested it with v0.3.0

**Comment by Normvariante** [comment]
I changed the `VisitString` function so that it replaces line breaks:

    function VisitString {

        param (
            $InputObject
        )

        Write-Verbose -Message "Visit string $InputObject";

        if ($InputObject -isnot [String]) {
            return $InputObject.ToString() -replace '\\', '\\' -replace "`r`n", "<br>"
        }

        return $InputObject -replace '\\', '\\' -replace "`r`n", "<br>"
    }

It does fix my example from above, but I'm not sure what else it might break :)

**Comment by BernieWhite** [comment]
@Normvariante Using normal markdown rules, I would suggest that a single line break should be omitted complete and not preserved.

i.e.

````
|String1|String2|String3|
| --- | --- | --- |
|1 2 3 4 5|One, Two,
Three|6 7 8 9 0|
````

would be

````
|String1|String2|String3|
| --- | --- | --- |
|1 2 3 4 5|One, Two, Three|6 7 8 9 0|

**Comment by Normvariante** [comment]
I'm not very experienced with using markdown, but according to this [stackoverflow thread](https://stackoverflow.com/questions/11700487/newline-in-markdown-table) it's a good idea to have a choice between `\` and `<br>`

**Comment by BernieWhite** [comment]
@Normvariante Unreleased at the moment, but to address your problem you will be able to use:

```powershell
Invoke-PSDocument .... -Option @{ 'Markdown.WrapSeparator' = '<br />' };
```

**Comment by Normvariante** [comment]
Hmm I'm not exactly sure what's going on, but I think now the `Section` is broken.

In the `VisitSection` function, `VisitString` is called like this:

    VisitString("`n$(''.PadLeft($section.Level, '#')) $($section.Content)");

The `` `n`` that is inserted before calling `VisitString` is being replaced by the wrap separator. For me that meant that sections that come right after tables are not rendered correctly. I changed the line to 

	"`n{0}" -f (VisitString("$(''.PadLeft($section.Level, '#')) $($section.Content)")); 

and that solved it for me.

**Comment by BernieWhite** [comment]
@normvariante are you able to provide an example?

**Comment by BernieWhite** [comment]
@Normvariante You are correct. The wrapping should have been limited, but I found a logic error that gave it a greater scope then intended. If you have an example that you know fails that would be helpful so that I can build a unit test to prevent regression.

**Comment by Normvariante** [comment]
Hi,

using ``"`n{0}" -f (VisitString("$(''.PadLeft($section.Level, '#')) $($section.Content)"));``:

    Document 'Test' {

        Title 'Working'

        Section 'Heading 1' {
        
            Get-Service | Select -First 3 -Property Name,Status, Starttype | Table

        }

        Section 'Heading 2' {
        
            Get-Service | Select -Last 3 -Property Name, Status, Starttype  | Table
        
        }

    }

    Invoke-PSDocument -Name 'Test' -PassThru 

Result is:

	# Working

	## Heading 1

	|Name|Status|StartType|
	| --- | --- | --- |
	|AJRouter|Stopped|Manual|
	|ALG|Stopped|Manual|
	|AppIDSvc|Stopped|Manual|

	## Heading 2

	|Name|Status|StartType|
	| --- | --- | --- |
	|XblGameSave|Stopped|Disabled|
	|XboxGipSvc|Stopped|Manual|
	|XboxNetApiSvc|Stopped|Disabled|

Rendered with mkdocs: https://i.imgur.com/OVyYQL4.png

using ``"(VisitString("`n$(''.PadLeft($section.Level, '#')) $($section.Content)"));``:

    Document 'Test' {

        Title 'Broken'

        Section 'Heading 1' {
        
            Get-Service | Select -First 3 -Property Name,Status, Starttype | Table

        }

        Section 'Heading 2' {
        
            Get-Service | Select -Last 3 -Property Name, Status, Starttype  | Table
        
        }

    }

    Invoke-PSDocument -Name 'Test' -PassThru 

Result is:

	# Broken
	## Heading 1

	|Name|Status|StartType|
	| --- | --- | --- |
	|AJRouter|Stopped|Manual|
	|ALG|Stopped|Manual|
	|AppIDSvc|Stopped|Manual|
	## Heading 2

	|Name|Status|StartType|
	| --- | --- | --- |
	|XblGameSave|Stopped|Disabled|
	|XboxGipSvc|Stopped|Manual|
	|XboxNetApiSvc|Stopped|Disabled|

Rendered with mkdocs: https://i.imgur.com/XYxw3Ss.png

There simple are new lines missing between the last row of the table and the sections. I tried to find out if this is invalid markdown syntax, but I didn't find anything about hit. However, you specifically add a new line in the visitSection function, so I assumed it is necessary. I only tested this with mkdocs.

**Comment by BernieWhite** [comment]
Released as [v0.4.0](https://github.com/BernieWhite/PSDocs/releases/tag/v0.4.0)
